src/main/java/net/fortuna/ical4j/model/component/Observance.java

```diff
@@ -155,12 +155,21 @@ public final Date getLatestOnset(final Date date) {
 
         if (initialOnset == null) {
             try {
-                initialOnset = applyOffsetFrom(calculateOnset(((DtStart) getProperty(Property.DTSTART)).getDate()));
+                DtStart dtStart = (DtStart) getProperty(Property.DTSTART);
+                if (dtStart == null) {
+                    throw new ConstraintViolationException("Missing DTSTART property");
+                }
+                initialOnset = applyOffsetFrom(calculateOnset(dtStart.getDate()));
             } catch (ParseException e) {
```

Comment:
This is a Java 7 language construct, so will make this change after the 2.0 release when we drop support for Java 6.